libs/postgres_backend/src/lib.rs

```diff
@@ -416,15 +433,11 @@ impl<IO: AsyncRead + AsyncWrite + Unpin> PostgresBackend<IO> {
             _ = shutdown_watcher() => {
                 // We were requested to shut down.
                 tracing::info!("shutdown request received during handshake");
-                return Ok(())
+                return Err(QueryError::Shutdown)
             },
 
-            result = self.handshake(handler) => {
-                // Handshake complete.
```

Comment:
Note to reviewers: this check gets dropped because handshake() is now returning a Disconnected error in the case where there was an EOF during handshake.